src/shader/shaderLightSampling.js

```diff
@@ -1,11 +1,51 @@
 export const shaderLightSampling = /* glsl */`
 
+float getSpotAttenuation( const in float coneCosine, const in float penumbraCosine, const in float angleCosine ) {
+
+	return smoothstep( coneCosine, penumbraCosine, angleCosine );
+
+}
+
+float getDistanceAttenuation( const in float lightDistance, const in float cutoffDistance, const in float decayExponent ) {
+
+		// based upon Frostbite 3 Moving to Physically-based Rendering
+		// page 32, equation 26: E[window1]
+		// https://seblagarde.files.wordpress.com/2015/07/course_notes_moving_frostbite_to_pbr_v32.pdf
+		float distanceFalloff = 1.0 / max( pow( lightDistance, decayExponent ), EPSILON );
+
+		if ( cutoffDistance > 0.0 ) {
+			distanceFalloff *= pow2( saturate( 1.0 - pow4( lightDistance / cutoffDistance ) ) );
+		}
+
+		return distanceFalloff;
```

Comment:
Looks like some off indentation here

Reply:
Got it, it's a shame we can't lint these also. If they were loaded as files (with some mechanism for includes / templating) that could prevent these from making it past the build tests. Not sure if it's worth it.

Reply:
Ideally there'd be some way to lint content in these template tags - I'll have to look around for something like that. Worst cast we _could_ write a script to pull out the content from these strings, write it to a file, and lint it there but there'd be a lot of indirection in the results...